webplugin/js/app/kommunicate-ui.js

```diff
@@ -404,14 +404,46 @@ setAvailabilityStatus : function (status){
 showClosedConversationBanner  : function(isConversationClosed){
     var messageText = MCK_LABELS["closed.conversation.message"];
     var conversationStatusDiv = document.getElementById("mck-conversation-status-box");
-    if (isConversationClosed){
+    var isCSATenabled = kommunicate._globals.collectFeedback;
+    var FEEDBACK_UPDATE_URL = "/feedback"
+    isConversationClosed && kommunicateCommons.modifyClassList( {class : ["mck-box-form"]}, "n-vis");
+    if(isCSATenabled && isConversationClosed && !kommunicateCommons.isConversationClosedByBot()){
+        mckUtils.ajax({
+            type: 'GET',
+            url: Kommunicate.getBaseUrl() + FEEDBACK_UPDATE_URL + '/'+ CURRENT_GROUP_DATA.tabId ,
+            global: false,
+            contentType: 'application/json',
+            success: function (data) {
+                CURRENT_GROUP_DATA.currentGroupFeedback = data.data;
+                kommunicateCommons.modifyClassList( {class : ["mck-box-form"]}, "n-vis");
+                kommunicateCommons.modifyClassList( {class : ["mck-csat-text-1"]}, "","n-vis");
+                kommunicateCommons.modifyClassList( {id : ["mck-sidebox-ft"]}, "mck-closed-conv-banner");
+                if( data.data && data.data.rating && data.data.comments.length> 0){
```

Comment:
@akshatdhabi don't you think that the entire code inside the `success` of the API call should be wrapped inside the `if(data.data && data.data.rating)` check?

Also, I think there should be an `error: function() { ... }` to handle any errors thrown by the API or code.

Reply:
No it will not be said `if(data.data && data.data.rating)` check as to when the data comes as null we will trigger the first step of CSAT rating